### ISSUE TITLE
Broken login functionality in TYPO3 11 LTS

### ISSUE THREAD

**Issue by tobibuenter** [marketing-factory/oauth2#31]
We created a small extension with just the following ext_localconf.php content
```
\Mfc\OAuth2\ResourceServer\Registry::addServer(
        'gitlab',
        'Login mit GitLab',
        \Mfc\OAuth2\ResourceServer\GitLab::class,
        [
            'enabled'   => true,
            'arguments' => [
                'appId'                => $extensionConfiguration['appId'],
                'appSecret'            => $extensionConfiguration['appSecret'],
                'gitlabServer'         => 'OUR-OWN-GITLAB-SERVER',
                'gitlabAdminUserLevel' => \Mfc\OAuth2\ResourceServer\GitLab::USER_LEVEL_MAINTAINER,
                'gitlabDefaultGroups'  => $extensionConfiguration['defaultGroups'],
                'gitlabUserOption'     => 3,
                'blockExternalUser'    => true,
                'projectName'          => 'PROJECT-NAME',
            ],
        ]
    );
```

I debugged the 3 used variables, all contains the right values.

`[BE][cookieSameSite] is set to "lax"`

I also set this information in our gitlab instance.
```
When creating the application within GitLab, you might need the following information:
- Redirect URI: `<your-domain-here>/typo3/index.php`
- Scopes: `api`,`read_user`,`openid`
```

Login startscreen works like expected in TYPO3 11.
<img width="305" alt="Bildschirmfoto 2022-01-17 um 08 20 49" src="https://user-images.githubusercontent.com/19902193/149738824-d5c87275-f8db-4a45-962f-5f986ffe5e0c.png">

If I switch to “login with OAuth”, the login form looks pretty wrong.
<img width="315" alt="Bildschirmfoto 2022-01-17 um 08 20 58" src="https://user-images.githubusercontent.com/19902193/149738866-09cb78fd-d8cd-45aa-b21d-3d54061e3bac.png">

If I just hit the "Anmeldung"-button, there is no redirect to gitlab, nothing. The login-screen just get reloaded. Tested in several browsers on the latest Mac OS.

In TYPO3 10 all works pretty fine with the same little extension.

**Environment:** 

- Docker with ddev
- PHP 7.4
- TYPO3 11 LTS
- OAuth Version 3.0.0

**Comment by zenobio93** [comment]
Should be fixed now within #30 and published as new version 3.0.1

**Comment by tobibuenter** [comment]
GREAT! Thank you very much for your very fast fix!!
